druntime/src/etc/linux/memoryerror.d

```diff
@@ -22,14 +27,68 @@ version (CRuntime_Glibc)
         version = MemoryErrorSupported;
 }
 
+private __gshared ubyte[SIGSTKSZ] altStack;
+
+/**
+Registers a signal handler for SIGSEGV that turns them into an assertion failure,
+providing a more descriptive error message and stack trace if the program is
+compiled with debug info and D assertions (as opposed to C assertions).
+
+Differences with version 1 are:
+- The handler is registered with SA_ONSTACK, so it can handle stack overflows.
+- It uses `assert(0)` instead of `throw new Error` and doesn't support catching the error.
+
+Returns: whether the registration was successful
+*/
+bool registerMemoryErrorHandler2()
```

Comment:
2 in a name is a massive ick.

Reply:
I considered naming it `registerMemoryAssertHandler` as well, but I thought the `2` would express its relation with the older function well. I don't know what's so icky about it, but I don't mind changing it.